Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using System.Text;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics
+{
+    public interface IDiagnosticsLogger
+    {
+        Task<IHttpResponse> LogDiagnosticsData(string eventType, string message);
+    }
+    public class DiagnosticsLogger : IDiagnosticsLogger
+    {
+        private readonly IHttpClient httpClient;
+        private readonly IServicesConfig servicesConfig;
+              
+        public DiagnosticsLogger(IHttpClient httpClient, IServicesConfig servicesConfig)
+        {
+            this.httpClient = httpClient;
+            this.servicesConfig = servicesConfig;
+        }
+
+        public async Task<IHttpResponse> LogDiagnosticsData(string eventType, string message = "")
```

Comment:
please add the same `Caller...` params used in `Logger.cs` to capture information about the filename where the event is coming from. Hopefully the message is unique enough, but I would include `callerName`, `filePath` and `lineNumber` at the end of the msg, e.g. in brackets